files/adfc-hh-connect.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+# Aufbau einer VPN-Verbindung zum ADFC-HH UCS-MASTER
+
+
+# Uptime & Hostname
+UT=$(uptime | cut -d ' ' -f 4)
+HOST=$(hostname)
+
+
+# Alte Fehlermeldungen löschen
+find /$HOST* -mtime 1 -exec rm {} \;
+
+
+# Herstellen der VPN-Verbindung
+vpnc adfc-hh.conf
+sleep 5
+
+
+# Test und Fehlerbehandlung, falls VPN-Tunnel nicht aufgebaut wurde
+ping -c 3 192.168.123.32 #2>&1> /dev/null
+if [ $? -ne 0 ]; then 
+# Fehler bei Verbindungsaufbau aufgetreten
+    vpnc-disconnect
+    sleep 5
+    killall vpnc
+    sleep 5
+# Kein weiter Verbindungsversuch bei Mehrfachfehler
+    if [ -f "/$HOST-REBOOT" ]; then
+	touch /$HOST-VERBINDUNGSFEHLER
+        exit 1
+    fi
+
+# erneuter Versuch ...
+    vpnc adfc-hh.conf
+    sleep 5
+    ping -c 3 192.168.3.32 2>&1> /dev/null
+# Reboot wenn nicht erfolgreich
+    if [ $? -ne 0 ]; then 
+        if [ $UT -lt 5 ]; then
+           touch /$HOST-REBOOT
+           reboot
```

Comment:
Heißt aber auch, wenn meine Telko probleme hat, rebootet im Keller ständig die ADFC Maschine.
Wirkt für mich etwas wie, als hätten wir einen alten Windows-Rechner wo wir nicht wissen, warum er sich aufhängt und dann booten wir ihn lieber mal. Ich würde ja lieber den Fehler finden, warum er sich nicht zum ADFC verbinden kann. Dafür wäre eine Nachrricht gut.

Reply:
Das Verbindungs-Skript auf dem Client ist nun in der Lage, Logdaten per Mail an eine definierte Adresse zu senden.